pkg/cli/webhook/github.go

```diff
@@ -0,0 +1,172 @@
+package webhook
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/google/go-github/v43/github"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/cli/prompt"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/formatting"
+	"golang.org/x/oauth2"
+)
+
+const apiPublicURL = "https://api.github.com/"
+
+type gitHubWebhookConfig struct {
+	Client              *github.Client
+	ControllerURL       string
+	RepoOwner           string
+	RepoName            string
+	WebhookSecret       string
+	PersonalAccessToken string
+	APIURL              string
+}
+
+func (w Webhook) githubWebhook(ctx context.Context) (*response, error) {
+	msg := "Would you like me to configure a GitHub Webhook for your repository? "
+	var configureWebhook bool
+	if err := prompt.SurveyAskOne(&survey.Confirm{Message: msg, Default: true}, &configureWebhook); err != nil {
+		return nil, err
+	}
+	if !configureWebhook {
+		return &response{UserDeclined: true}, nil
+	}
+
+	ghWebhook, err := askGHWebhookConfig(w.RepositoryURL, w.ControllerURL)
+	if err != nil {
+		return nil, err
+	}
+	ghWebhook.APIURL = w.ProviderAPIURL
+
+	return &response{
+		UserDeclined:        false,
+		ControllerURL:       ghWebhook.ControllerURL,
+		PersonalAccessToken: ghWebhook.PersonalAccessToken,
+		WebhookSecret:       ghWebhook.WebhookSecret,
+	}, ghWebhook.create(ctx)
+}
+
+func askGHWebhookConfig(repoURL, controllerURL string) (*gitHubWebhookConfig, error) {
+	gh := &gitHubWebhookConfig{}
+
+	var repo, defaultRepo string
+	if repoURL != "" {
+		if repo, _ := formatting.GetRepoOwnerFromGHURL(repoURL); repo != "" {
+			defaultRepo = repo
+		}
+	}
+	if defaultRepo != "" {
+		msg := fmt.Sprintf("Please enter the repository you want to be configured (default: %s):", defaultRepo)
+		if err := prompt.SurveyAskOne(&survey.Input{Message: msg}, &repo); err != nil {
+			return nil, err
+		}
+	} else {
+		msg := "Please enter the repository you want to be configured (eg. repo-owner/repo-name) : "
+		if err := prompt.SurveyAskOne(&survey.Input{Message: msg}, &repo,
+			survey.WithValidator(survey.Required)); err != nil {
+			return nil, err
+		}
+	}
+
+	if repo == "" {
+		repo = defaultRepo
+	}
+	repoArr := strings.Split(repo, "/")
+	if len(repoArr) != 2 {
+		return nil, fmt.Errorf("invalid repository, needs to be of format 'org-name/repo-name'")
+	}
+
+	gh.RepoOwner = repoArr[0]
+	gh.RepoName = repoArr[1]
+
+	if controllerURL == "" {
+		if err := prompt.SurveyAskOne(&survey.Input{
+			Message: "Please enter your controller public route URL: ",
+		}, &gh.ControllerURL, survey.WithValidator(survey.Required)); err != nil {
+			return nil, err
+		}
+	}
+
+	if err := prompt.SurveyAskOne(&survey.Input{
```

Comment:
I think there is another prompt type or way to hide the token/password on the console